src/pinwheel-wrapper.android.tsx

```diff
@@ -16,28 +17,33 @@ const createFragment = (viewId: number) => {
         // @ts-ignore
         UIManager.RNTPinwheel.Commands.create.toString(),
         [viewId],
-        );
-    }
-    
-    const RNTPinwheelView = (props: RNTPinwheelProps) => {
-        const ref = useRef(null);
-        useEffect(() => {
-            setTimeout(() => {
-                const viewId = findNodeHandle(ref.current);
-                if (viewId) {
-                    createFragment(viewId);
-                }
-
-                // events
-                const { RNTPinwheelEvents } = NativeModules;
-                const eventEmitter = new NativeEventEmitter(RNTPinwheelEvents);
-                const eventListener = eventEmitter.addListener('PINWHEEL_EVENT', (event) => {
-                    props.onEvent(event);
-                });
-            }, 10);
-        }, []);
-        
-        return <RNTPinwheel {...props} onEvent={(event) => {console.log`rawbee: ${JSON.stringify(event)}`}} ref={ref} />;
-    }
-    
-    export default RNTPinwheelView;
\ No newline at end of file
+    );
+}
+
+const RNTPinwheelView = (props: RNTPinwheelProps) => {
+    const ref = useRef(null);
+    const eventListenerRef = useRef<EmitterSubscription | null>(null);
+    useEffect(() => {
+        setTimeout(() => {
+            const viewId = findNodeHandle(ref.current);
+            if (viewId) {
+                createFragment(viewId);
+            }
+
+            // events
+            const {RNTPinwheelEvents} = NativeModules;
+            const eventEmitter = new NativeEventEmitter(RNTPinwheelEvents);
+            eventListenerRef.current = eventEmitter.addListener('PINWHEEL_EVENT', (event) => {
+                props.onEvent(event);
+            });
+        }, 10);
+
+        return () => {
+            eventListenerRef.current?.remove();
```

Comment:
[q] What's the purpose of this addition/changes here?

Reply:
This removes the event listener when the component is unmounted, in order to prevent leaking memory.